src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/ControlProblemElements.h

```diff
@@ -0,0 +1,745 @@
+                              const iDynTree::Vector3& position);
+
+    /**
+     * Set the desired trajectory
+     * @param acceleration  desired angular acceleration
+     * @param velocity desired  angular velocity
+     * @param rotation desired rotation matrix
+     * @throw std::runtime_error if the Type is different from POSE and ORIENTATION
+     */
+    void setDesiredTrajectory(const iDynTree::Vector3& acceleration,
+                              const iDynTree::Vector3& velocity,
+                              const iDynTree::Rotation& rotation);
+
+    /**
+     * Set the desired trajectory
+     * @param acceleration desired spatial acceleration
+     * @param velocity angular desired spatial velocity
+     * @param transform desired homogeneous transform
+     * @throw std::runtime_error if the Type is different from POSE
+     */
+    void setDesiredTrajectory(const iDynTree::SpatialAcc& acceleration,
+                              const iDynTree::Twist& velocity,
+                              const iDynTree::Transform& transform);
+
+    /**
+     * Set the desired trajectory
+     * @param acceleration desired acceleration
+     * @param velocity angular desired velocity
+     * @param transform desired position
+     * @throw std::runtime_error if the Type is different from ONE_DIMENSION
+     */
+    void setDesiredTrajectory(const double& acceleration,
+                              const double& velocity,
+                              const double& position);
+
+    /**
+     * Set the linear PID gains
+     * @param kp proportional gain
+     * @param kd derivative gain
+     * @throw std::runtime_error if the Type is different from POSE and POSITION
+     */
+    void setLinearPIDGains(const iDynTree::Vector3& kp, const iDynTree::Vector3& kd);
+
+    /**
+     * Set the orientation PID gains
+     * @param c0 c0 gain
+     * @param c1 derivative gain
+     * @param c2 proportional gain
+     * @throw std::runtime_error if the Type is different from ORIENTATION and POSITION
+     */
+    void setOrientationPIDGains(const double& c0, const double& c1, const double& c2);
+
+    /**
+     * Set the linear PID gains
+     * @param kp proportional gain
+     * @param kd derivative gain
+     * @throw std::runtime_error if the Type is different from ONE_DIMENSION
+     */
+    void setOneDegreePIDGains(const double& kp, const double& kd);
+
+    /**
+     * Get (and compute) the element matrix
+     * @return the element matrix
+     */
+    virtual const iDynTree::MatrixDynSize& getA() final;
+
+    /**
+     * Get (and compute) the element vector
+     * @return the element vector
+     */
+    virtual const iDynTree::VectorDynSize& getB() final;
+
+    /**
+     * Set if the link is in contact with the environment
+     * @param isInContact true if the link associated to the frame is in contact with the
+     * environment
+     */
+    void isInContact(bool isInContact);
```

Comment:
I am not convinced why this method should be in this class.

Reply:
If the link is in contact with the environment (and we don't want to break the contact) the desired acceleration should be zero. When `isInContact` is `true`, the desire acceleration is equal to zero. 

https://github.com/dic-iit/bipedal-locomotion-controllers/blob/ace9d4a5b7b432d3aee4cbe66077e298460c0301/src/OptimalControlUtilities/src/ControlProblemElements.cpp#L262-L277